### PR TITLE
fix(media): resolve managed-proxy context atomically in image-credentials

### PR DESCRIPTION
**Pull request by siddseethepalli**
## Summary

Follow-up to #27525 addressing reviewer feedback.

- **TOCTOU fix**: `resolveImageGenCredentials` previously called `buildManagedBaseUrl()` and `resolveManagedProxyContext()` as two separate async lookups. Between the two awaits the assistant API key could be revoked/cleared, so the function could return managed-proxy credentials with an empty `assistantApiKey` — downstream 401s instead of the polished "log in to Vellum" hint. Resolve the context once and derive `baseUrl` from the same snapshot using `MANAGED_PROVIDER_META` directly, and require `assistantApiKey` to be non-empty before returning managed credentials.
- **JSDoc cleanup**: Rewrite the module header to describe current behavior instead of narrating history ("previously duplicated across..."), per AGENTS.md comments-describe-current-state rule.
- Add a regression test covering the case where the platform URL is present but the assistant API key is empty (previously the 'success path with empty key' outcome).

## Test plan

- [x] `bun test src/__tests__/image-credentials.test.ts` passes (7 tests, including new TOCTOU-safe case)
- [x] `bunx tsc --noEmit` clean for `assistant/src/media/image-credentials.ts` and its test
- [x] Lint + formatting clean via pre-commit hook

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

<a href="https://app.devin.ai/review/vellum-ai/vellum-assistant/pull/27648" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

**Comment by siddseethepalli** [comment]
Addressed in #27662 (merged earlier today) — test file already updated with the correct expected URL and retitled.

File: assistant/src/media/image-credentials.ts [hunk 1]
```diff
@@ -33,19 +30,27 @@ export async function resolveImageGenCredentials(opts: {
   const { provider, mode } = opts;
 
   if (mode === "managed") {
-    const baseUrl = await buildManagedBaseUrl(provider);
-    if (!baseUrl) {
+    // Resolve platform URL + assistant API key from a single snapshot so
+    // baseUrl and assistantApiKey can't diverge if the credential is cleared
+    // between lookups.
+    const meta = MANAGED_PROVIDER_META[provider];
```
[the remark this answers is not in the document]
Comment (siddseethepalli) [reply assistant/src/media/image-credentials.ts:36]: Addressed in #27662
